test/extensions/transport_sockets/proxy_protocol/proxy_protocol_integration_test.cc

```diff
@@ -231,5 +232,171 @@ TEST_P(ProxyProtocolIntegrationTest, TestV2ProxyProtocol) {
+  ProxyProtocolHttpIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam()) {}
+
+  void TearDown() override {
+    test_server_.reset();
+    fake_upstream_connection_.reset();
+    fake_upstreams_.clear();
+  }
+
+  void setup(envoy::config::core::v3::ProxyProtocolConfig_Version version, bool health_checks,
+             std::string inner_socket) {
+    version_ = version;
+    health_checks_ = health_checks;
+    inner_socket_ = inner_socket;
+  }
+
+  void initialize() override {
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      auto* transport_socket =
+          bootstrap.mutable_static_resources()->mutable_clusters(0)->mutable_transport_socket();
+      transport_socket->set_name("envoy.transport_sockets.upstream_proxy_protocol");
+      envoy::config::core::v3::TransportSocket inner_socket;
+      inner_socket.set_name(inner_socket_);
+      envoy::config::core::v3::ProxyProtocolConfig proxy_proto_config;
+      proxy_proto_config.set_version(version_);
+      envoy::extensions::transport_sockets::proxy_protocol::v3::ProxyProtocolUpstreamTransport
+          proxy_proto_transport;
+      proxy_proto_transport.mutable_transport_socket()->MergeFrom(inner_socket);
+      proxy_proto_transport.mutable_config()->MergeFrom(proxy_proto_config);
+      transport_socket->mutable_typed_config()->PackFrom(proxy_proto_transport);
+
+      if (health_checks_) {
+        auto* cluster = bootstrap.mutable_static_resources()->mutable_clusters(0);
+        cluster->set_close_connections_on_host_health_failure(false);
+        cluster->mutable_common_lb_config()->mutable_healthy_panic_threshold()->set_value(0);
+        cluster->add_health_checks()->mutable_timeout()->set_seconds(20);
+        cluster->mutable_health_checks(0)->mutable_reuse_connection()->set_value(true);
+        cluster->mutable_health_checks(0)->mutable_interval()->set_seconds(1);
+        cluster->mutable_health_checks(0)->mutable_no_traffic_interval()->set_seconds(1);
+        cluster->mutable_health_checks(0)->mutable_unhealthy_threshold()->set_value(1);
+        cluster->mutable_health_checks(0)->mutable_healthy_threshold()->set_value(1);
+        cluster->mutable_health_checks(0)->mutable_http_health_check()->set_codec_client_type(
+            envoy::type::v3::HTTP1);
+        cluster->mutable_health_checks(0)->mutable_http_health_check()->set_path("/healthcheck");
+      }
+    });
+    BaseIntegrationTest::initialize();
+  }
+
+private:
+  envoy::config::core::v3::ProxyProtocolConfig_Version version_;
+  bool health_checks_;
+  std::string inner_socket_;
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, ProxyProtocolHttpIntegrationTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+// Test sending proxy protocol over http
+TEST_P(ProxyProtocolHttpIntegrationTest, TestV1ProxyProtocol) {
```

Comment:
we didn't do this for TCP, but for HTTP we should reuse the stream
Can you make sure sending a subsequent request on the client connection results in a second request on the upstream connection?

Reply:
Hey, could you clarify a bit more please? On line 332, we make a second request with the client and then verify the upstream doesn't get the header the second time. What you're suggesting to test is unrelated to that?

Reply:
oops, my bad! I was searching for raw HTTP and missed the reuse of the predefined string.  You're good here.